src/value.rs

```diff
@@ -91,6 +91,35 @@ impl Value {
             .count()
     }
 
+    /// Check if the value is a unit.
+    pub fn is_unit(&self) -> bool {
+        matches!(self, Value::Unit)
+    }
+
+    /// Unwrap a left sum value and return `None` on other values.
+    pub fn unwrap_left(&self) -> Option<Arc<Value>> {
```

Comment:
In e10f2df43b42b5865f5cd7be6ecd11d3f52cb498:

`unwrap_` is a super confusing name for a method that returns an `Option`. I would suggest calling it `to_left`.

You could also return an `Option<&Arc<Value>>`, avoid the `Arc::clone`, and call it `as_left`.

Reply:
I think we should rename the existing `split_sum` and `split_product` to `as_sum` and `as_product` as well. (Or `to_sum`/`to_product` if we are returning non-borrowed Arcs.)